### PR TITLE
ISPN-7963 Relocate Infinispan Hibernate cache provider

### PR DESCRIPTION
**Pull request by galderz**
Relocate Infinispan Hibernate cache provider, along with all its related history to Infinispan repository.

Let's get this in. These are the TODOs to follow in future commits/JIRAs:

- [ ] Address the rest of @rvansa's comments in #5254 
- [ ] Make the hibernate-cache pom.xml produce OSGI bundles.
- [ ] Port to 9.0.x branch.
- [ ] Make the corresponding changes in the Hibernate git repository to remove the cache provider and add artifact relocation details.
- [ ] Create simple tutorial(s).
- [ ] Update documentation (if necessary).

### CODE REVIEW

**Comment by galderz** [comment]
CI shows two test failures here:

```
org.infinispan.test.hibernate.cache.functional.BulkOperationsTest.testBulkOperations[read-write, DIST_SYNC]
org.infinispan.test.hibernate.cache.timestamp.TimestampsRegionImplTest.testEvict[JTA, INVALIDATION_SYNC,AccessType[transactional]]
```

**Comment by galderz** [comment]
I've been running locally wo/ issues, so probably those failures are due to running with 1 CPU or something like that. I'll try to replicate in my local linux server tomorrow. Let's not halt the integration cos of those failures. I'll address them tomorrow or when I come back from vacation (2 weeks)

**Comment by tristantarrant** [comment]
Merged to master. Thanks @galderz

**Comment by galderz** [comment]
Thanks @tristantarrant! I'll create JIRAs tomorrow for all remaining TODOs, including these random test failures.

**Comment by galderz** [comment]
I've created JIRAs for all remaining todos (see description). All these JIRAs are marked as following up on [ISPN-7963](https://issues.jboss.org/browse/ISPN-7963).

I've also created a JIRA to port over the Hibernate cache provider to 9.0.x ([ISPN-8028](https://issues.jboss.org/browse/ISPN-8028)), which should be done once all follow up JIRAs that affect the code have been completed.

File: hibernate-cache/src/main/java/org/infinispan/hibernate/cache/access/LockingInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ * {@link CompletableFuture} and we'll wait for it here.
+ */
+public class LockingInterceptor extends NonTransactionalLockingInterceptor {
+	private static final Log log = LogFactory.getLog(LockingInterceptor.class);
+
+   protected final InvocationFinallyAction unlockAllReturnCheckCompletableFutureHandler = new InvocationFinallyAction() {
+      @Override
+      public void accept(InvocationContext rCtx, VisitableCommand rCommand, Object rv, Throwable throwable) throws Throwable {
+         lockManager.unlockAll(rCtx);
+         if (rv instanceof CompletableFuture) {
```
Comment (rvansa) [inline hibernate-cache/src/main/java/org/infinispan/hibernate/cache/access/LockingInterceptor.java]: `return rv instanceof CompletableFuture ? ((CompletableFuture) rv).join() : rv` and change this anonymous class to lambda implementing InvocationFinally**Function**

**Pull request merged**